voyager-core/src/commonMain/kotlin/cafe/adriel/voyager/core/lifecycle/NavigatorScreenLifecycle.kt

```diff
@@ -4,8 +4,8 @@ import androidx.compose.runtime.ProvidableCompositionLocal
 import androidx.compose.runtime.staticCompositionLocalOf
 import cafe.adriel.voyager.core.screen.Screen
 
-public val LocalNavigatorScreenLifecycleProvider: ProvidableCompositionLocal<NavigatorScreenLifecycleProvider> =
-    staticCompositionLocalOf { DefaultNavigatorScreenLifecycleProvider() }
+public val LocalNavigatorScreenLifecycleProvider: ProvidableCompositionLocal<NavigatorScreenLifecycleProvider?> =
+    staticCompositionLocalOf { null }
```

Comment:
Any particular reason for making it nullable and initialising it as null as well?

Reply:
I have moved the default implementation to the `Navigator` module, when this is null, it will use the default implementation. The reason is for removing the android lifecycle API from the core module and move to the navigator, also avoid cycle dependencies